library/src/main/java/com/google/android/exoplayer/text/CuePainter.java

```diff
@@ -350,4 +362,164 @@ private static boolean areCharSequencesEqual(CharSequence first, CharSequence se
     return first == second || (first != null && first.equals(second));
   }
 
+  /**
+   * Gets position anchor given an anchor and an alignment value. This method takes into
+   * account that the parameter position anchor could be not set and then anchor "inherits" its
```

Comment:
Sounds good. I think this was wrong in the previous version so this failed in some of my tests and I added this behavior here. Anyway, as you suggest, WebVttParser is a better way for implementing that, as that's very specifc of WebVTT.

Just added two changes:
 1.- What you suggest. Removing inheritance behavior from CuePainter.
 2.- Modified WebVTTParser so whenever an alignment is not defined, alignmentToAnchor method returns MIDDLE alignment (that's what WebVTT spec says).